src/operator/mergeMap.ts

```diff
@@ -17,34 +17,41 @@ import {InnerSubscriber} from '../InnerSubscriber';
  * @returns {Observable} an Observable that emits the result of applying the transformation function to each item
  * emitted by the source Observable and merging the results of the Observables obtained from this transformation
  */
-export function mergeMap<T, R, R2>(project: (value: T, index: number) => Observable<R>,
-                                   resultSelector?: (outerValue: T, innerValue: R, outerIndex: number, innerIndex: number) => R2 | number,
-                                   concurrent: number = Number.POSITIVE_INFINITY): Observable<R2> {
+export function mergeMap<T, I, R>(project: (value: T, index: number) => ObservableInput<I>,
+                                  resultSelector?: (outerValue: T, innerValue: I, outerIndex: number, innerIndex: number) => R | number,
```

Comment:
Grouping issue again; I think this should be `((outerValue: T, innerValue: I, outerIndex: number, innerIndex: number) => R) | number`, which makes `mergeMap` assignable to its signature.